core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceAggregationBuilder.java

```diff
@@ -82,6 +82,23 @@ public GeoDistanceAggregationBuilder(StreamInput in) throws IOException {
         unit = DistanceUnit.readFromStream(in);
     }
 
+    // for parsing
+    GeoDistanceAggregationBuilder(String name) {
```

Comment:
You can use `ContstructingObjectParser` so you don't need to make these funny sacrifices. It is built for exactly these sorts of situations. It isn't always the right thing, but sometimes it is. Maybe here?